lametro/models/cms.py

```diff
@@ -24,6 +26,64 @@ class AboutPage(Page):
     ]
 
 
+class BoardMemberDetails(
+    DraftStateMixin, RevisionMixin, PreviewableMixin, models.Model
+):
+    include_in_dump = True
+
+    class Meta:
+        verbose_name_plural = "Board Member Details"
+
+    person = models.OneToOneField(
+        "lametro.LAMetroPerson", on_delete=models.CASCADE, related_name="details"
+    )
+    headshot = models.ForeignKey(
+        "wagtailimages.Image",
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name="+",
+    )
+    headshot_source = models.CharField(
+        max_length=256, blank=True, null=True, default="Metro"
+    )
+    bio = RichTextField(blank=True, null=True)
+    _revisions = GenericRelation(
+        "wagtailcore.Revision", related_query_name="member_details"
+    )
+
+    @property
+    def revisions(self):
+        return self._revisions
+
+    def get_url(self):
+        return reverse("lametro:person", kwargs={"slug": self.person.slug})
+
+    def __str__(self):
+        loaded_obj = (
+            type(self)
+            .objects.select_related("person")
+            .prefetch_related("person__memberships")
+            .get(id=self.id)
+        )
+        return f"{loaded_obj.person.name}{' (current)' if loaded_obj.person.current_memberships.exists() else ''}"
+
+    def get_preview_context(self, request, mode_name):
+        context = super().get_preview_context(request, mode_name)
+        context["person"] = self.person
+        context["person_details"] = self
+
+        council_post = self.person.latest_council_membership.post
+        context["qualifying_post"] = council_post.acting_label
+
+        context["preview"] = True
```

Comment:
The map and committee/board report modules require a bunch of extra context, so I use this to purposefully omit them from page previews. (I also added a note to the admin interface.)